docs/create-packages/native-files-in-net-packages.md

```diff
@@ -0,0 +1,165 @@
+
+### Runtime assets
+
+Runtime assets are .NET managed assemblies copied to the output directory on build and publish, and the .NET SDK generates the `deps.json` with all the RID-specific asset information.
+NuGet will select runtime assets from `runtimes/{rid}/lib/{tfm}/`, and if that doesn't exist, then it will fall back to `lib/{tfm}/`.
+The best `{rid}` directory is selected first, followed by `{tfm}`.
+However, in this scenario of creating a .NET package with native libraries, it is recommended to use `runtimes/` because if `lib/` is used, NuGet will think the package is compatible with projects using `packages.config`, but those projects will fail at runtime because the native assets will not be copied.
+
+### Native assets
+
+Native assets are also copied to the output directory on build and publish, and are present in the `deps.json` file.
+NuGet will select native assets from the `runtimes/{rid}/native/` directory.
+
+> [!NOTE]
+> The [.NET SDK flattens any directory structure](https://github.com/dotnet/sdk/issues/9643) under `runtimes/{rid}/native/` when copying to the output directory.
+
+## Putting it together
+
+How to build any binaries native libraries is out of scope of this document.
+It is assumed that all the native binaries and the .NET assemblies have been completed and copied to a machine for packing.
+
+When packing with [NuGet's MSBuild Pack target](../reference/msbuild-targets.md#pack-target), you can include arbitrary files in arbitrary package paths using `Pack="true" PackagePath="{path}"` metadata on MSBuild items.
+For example, `<None Include="../cross-compile/linux-x64/libcontoso.so" Pack="true" PackagePath="runtimes/linux-x64/native/" />`.
+
+Consider a package `Contoso.Native` that provides .NET APIs to `libcontoso.dll` on Windows, `libcontoso.dylib` on OSX, and `libcontoso.so` on Linux, using `[DllImport("contoso")]`.
+The package contents should be (excluding files required by the [Open Packaging Conventions](https://en.wikipedia.org/wiki/Open_Packaging_Conventions))
+
+### Example 1
+
+If `Contoso.Native` is built as AnyCPU:
+
+```text
+Contoso.Native.1.0.0.nuspec
+ref/net8.0/Contoso.Native.dll
+runtimes/any/lib/net8.0/Contoso.Native.dll
+runtimes/linux-arm64/native/libcontoso.so
+runtimes/linux-x64/native/libcontoso.so
+runtimes/osx-arm64/native/libcontoso.dylib
+runtimes/osx-x64/native/libcontoso.dylib
+runtimes/win-arm64/native/contoso.dll
+runtimes/win-x64/native/contoso.dll
+```
+
+While the `ref/` and `runtimes/any/lib` copies of `Contoso.Native.dll` could be deduplicated into a single `lib/net8.0/Contoso.Native.dll` file, this is not recommended because NuGet will believe this package is compatible with projects using `packages.config`, and these projects will fail at runtime when the native `contoso.dll` is not found.
+
+### Example 2
```

Comment:
Another possible variation - `Contoso.Native` compiled differently for different OS, but for each not architecture of that OS. So it could be:

```text
runtimes/linux/lib/net8.0/Contoso.Native.dll
runtimes/linux-arm64/native/libcontoso.so
runtimes/linux-x64/native/libcontoso.so
runtimes/osx/lib/net8.0/Contoso.Native.dll
runtimes/osx-arm64/native/libcontoso.dylib
runtimes/osx-x64/native/libcontoso.dylib
runtimes/win/lib/net8.0/Contoso.Native.dll
runtimes/win-arm64/native/contoso.dll
runtimes/win-x64/native/contoso.dll
```